test/unit/test_sdb.c

```diff
@@ -41,6 +41,25 @@ bool test_sdb_list_delete(void) {
 	mu_end;
 }
 
+static int __cmp_asc(const void *a, const void *b) {
+	const SdbKv *ka = a;
+	const SdbKv *kb = b;
+	return strcmp (ka->key, kb->key);
+}
+
+bool test_sdb_list_big(void) {
+	Sdb *db = sdb_new (NULL, NULL, false);
+	int i;
+	for ( i=0; i < 1000000; i++) {
+    	sdb_num_set (db, sdb_fmt (0, "%d", i), i + 1, 0);
+	}
+	SdbList *list = sdb_foreach_list (db, true);
+	ls_sort (list, __cmp_asc);
+	// TODO: verify if its sorted
```

Comment:
Here you are sorting twice since sdb_foreach_list with the true flag already does the same operation as above.